### ISSUE TITLE
Carddav always returns a 403 saying sync token is invalid or unknown

### ISSUE THREAD

**Issue by binarykitchen** [owncloud/core#30820]
Already searched and found no answer. Spotted this line in the log multiple times over again:

```
{"reqId":"FHyGsrrc72LGHjvRvnwY","level":4,"time":"2018-03-19T20:34:08+00:00","remoteAddr":"113.197.65.22","user":"michael.heuberger","app":"carddav","method":"REPORT","url":"\/remote.php\/carddav\/addressbooks\/michael.heuberger\/contacts\/","message":"Exception: {\"Message\":\"HTTP\\\/1.1 403 Invalid or unknown sync token\",\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\InvalidSyncToken\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Sync\\\/Plugin.php(65): Sabre\\\\DAV\\\\Sync\\\\Plugin->syncCollection('addressbooks\\\/mi...', Object(Sabre\\\\DAV\\\\Xml\\\\Request\\\\SyncCollectionReport))\\n#1 [internal function]: Sabre\\\\DAV\\\\Sync\\\\Plugin->Sabre\\\\DAV\\\\Sync\\\\{closure}('{DAV:}sync-coll...', Object(Sabre\\\\DAV\\\\Xml\\\\Request\\\\SyncCollectionReport), 'addressbooks\\\/mi...')\\n#2 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Object(Closure), Array)\\n#3 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(718): Sabre\\\\Event\\\\EventEmitter->emit('report', Array)\\n#4 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpReport(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#5 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#6 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(479): Sabre\\\\Event\\\\EventEmitter->emit('method:REPORT', Array)\\n#7 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(254): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#8 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/appinfo\\\/v1\\\/carddav.php(85): Sabre\\\\DAV\\\\Server->exec()\\n#9 \\\/var\\\/www\\\/owncloud\\\/remote.php(165): require_once('\\\/var\\\/www\\\/ownclo...')\\n#10 {main}\",\"File\":\"\\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Sync\\\/Plugin.php\",\"Line\":123,\"User\":\"michael.heuberger\"}"}
```

My server config
[config_report_20180319 (1).json.txt](https://github.com/owncloud/core/files/1826910/config_report_20180319.1.json.txt)

Any clues what this could be?

**Comment by ownclouders** [comment]
[GitMate.io](https://gitmate.io) thinks possibly related issues are https://github.com/owncloud/core/issues/10783 (CardDAV Sync: Sabre\DAV\Exception: Invalid result from updateProperties), https://github.com/owncloud/core/issues/18409 (PROPPATCH returns 403), https://github.com/owncloud/core/issues/16541 (Contact sync CardDAV issues), and https://github.com/owncloud/core/issues/27920 (Home folder unavailable after upgrade til version 10).

**Comment by Perm1990** [comment]
Have same problem (?) with owncloud 10.0.10

**Comment by joostdekeijzer** [comment]
Same sync problem and error on macOS Mojave (10.14.1) and ownCloud 10.0.10 syncing cardDAV with Contacts app.

Also see https://central.owncloud.org/t/owncloud-10-0-9-mac-os-x-10-13-6-high-sierra-problem-sync-contacts-carddav-v/15621

**Comment by joostdekeijzer** [comment]
In my setup, the error is generated here:
https://github.com/sabre-io/dav/blob/3.2/lib/DAV/Sync/Plugin.php#L123

With modified logging, I see that `self::SYNCTOKEN_PREFIX` is 'http://sabre.io/ns/sync/' and `$syncToken` is '0ba8eaf8-0897-4a65-8e05-849145bfaf9a'

**Comment by joostdekeijzer** [comment]
After removing & adding the account (through macOS System Preferences > Accounts > cardDAV) the following happens in the logfile (debug level 0)

1. Add account (successfully)
```
{"reqId":"W@63iC0s2qdTNtrs6ztMwAAAAAM","level":0,"time":"2018-11-16T12:26:48+00:00","remoteAddr":"213.127.150.1","user":"--","app":"webdav","method":"OPTIONS","url":"\/remote.php\/dav\/principals\/users\/xxxxx","message":"Exception: HTTP\/1.1 401 No public access to this resource., No 'Authorization: Basic' header found. Either the client didn't send one, or the server is misconfigured: {\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\NotAuthenticated\",\"Message\":\"No public access to this resource., No 'Authorization: Basic' header found. Either the client didn't send one, or the server is misconfigured\",\"Code\":0,\"Trace\":\"#0 [internal function]: Sabre\\\\DAV\\\\Auth\\\\Plugin->beforeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#1 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#2 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(466): Sabre\\\\Event\\\\EventEmitter->emit('beforeMethod', Array)\\n#3 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(254): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#4 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/apps\\\/dav\\\/lib\\\/Server.php(298): Sabre\\\\DAV\\\\Server->exec()\\n#5 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/apps\\\/dav\\\/appinfo\\\/v2\\\/remote.php(31): OCA\\\\DAV\\\\Server->exec()\\n#6 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/remote.php(165): require_once('\\\/home\\\/xxxxx\\\/dom...')\\n#7 {main}\",\"File\":\"\\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Auth\\\/Plugin.php\",\"Line\":168}"}
{"reqId":"W@63iS0s2qdTNtrs6ztMxAAAAAk","level":0,"time":"2018-11-16T12:26:50+00:00","remoteAddr":"213.127.150.1","user":"--","app":"webdav","method":"PROPFIND","url":"\/remote.php\/dav\/addressbooks\/users\/xxxxx\/","message":"Exception: HTTP\/1.1 401 No public access to this resource., No 'Authorization: Basic' header found. Either the client didn't send one, or the server is misconfigured: {\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\NotAuthenticated\",\"Message\":\"No public access to this resource., No 'Authorization: Basic' header found. Either the client didn't send one, or the server is misconfigured\",\"Code\":0,\"Trace\":\"#0 [internal function]: Sabre\\\\DAV\\\\Auth\\\\Plugin->beforeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#1 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#2 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(466): Sabre\\\\Event\\\\EventEmitter->emit('beforeMethod', Array)\\n#3 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(254): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#4 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/apps\\\/dav\\\/lib\\\/Server.php(298): Sabre\\\\DAV\\\\Server->exec()\\n#5 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/apps\\\/dav\\\/appinfo\\\/v2\\\/remote.php(31): OCA\\\\DAV\\\\Server->exec()\\n#6 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/remote.php(165): require_once('\\\/home\\\/xxxxx\\\/dom...')\\n#7 {main}\",\"File\":\"\\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Auth\\\/Plugin.php\",\"Line\":168}"}
```

2. After some time (just waiting doing nothing)
```
{"reqId":"W@63qC0s2qdTNtrs6ztM5AAAABU","level":0,"time":"2018-11-16T12:27:21+00:00","remoteAddr":"213.127.150.1","user":"--","app":"webdav","method":"OPTIONS","url":"\/remote.php\/dav\/principals\/users\/xxxxx","message":"Exception: HTTP\/1.1 401 No public access to this resource., No 'Authorization: Basic' header found. Either the client didn't send one, or the server is misconfigured: {\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\NotAuthenticated\",\"Message\":\"No public access to this resource., No 'Authorization: Basic' header found. Either the client didn't send one, or the server is misconfigured\",\"Code\":0,\"Trace\":\"#0 [internal function]: Sabre\\\\DAV\\\\Auth\\\\Plugin->beforeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#1 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#2 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(466): Sabre\\\\Event\\\\EventEmitter->emit('beforeMethod', Array)\\n#3 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(254): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#4 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/apps\\\/dav\\\/lib\\\/Server.php(298): Sabre\\\\DAV\\\\Server->exec()\\n#5 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/apps\\\/dav\\\/appinfo\\\/v2\\\/remote.php(31): OCA\\\\DAV\\\\Server->exec()\\n#6 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/remote.php(165): require_once('\\\/home\\\/xxxxx\\\/dom...')\\n#7 {main}\",\"File\":\"\\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Auth\\\/Plugin.php\",\"Line\":168}"}
{"reqId":"W@63qi0s2qdTNtrs6ztM6AAAAAo","level":0,"time":"2018-11-16T12:27:22+00:00","remoteAddr":"213.127.150.1","user":"--","app":"webdav","method":"PROPFIND","url":"\/remote.php\/dav\/addressbooks\/users\/xxxxx\/","message":"Exception: HTTP\/1.1 401 No public access to this resource., No 'Authorization: Basic' header found. Either the client didn't send one, or the server is misconfigured: {\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\NotAuthenticated\",\"Message\":\"No public access to this resource., No 'Authorization: Basic' header found. Either the client didn't send one, or the server is misconfigured\",\"Code\":0,\"Trace\":\"#0 [internal function]: Sabre\\\\DAV\\\\Auth\\\\Plugin->beforeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#1 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#2 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(466): Sabre\\\\Event\\\\EventEmitter->emit('beforeMethod', Array)\\n#3 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(254): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#4 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/apps\\\/dav\\\/lib\\\/Server.php(298): Sabre\\\\DAV\\\\Server->exec()\\n#5 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/apps\\\/dav\\\/appinfo\\\/v2\\\/remote.php(31): OCA\\\\DAV\\\\Server->exec()\\n#6 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/remote.php(165): require_once('\\\/home\\\/xxxxx\\\/dom...')\\n#7 {main}\",\"File\":\"\\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Auth\\\/Plugin.php\",\"Line\":168}"}
{"reqId":"W@63rS0s2qdTNtrs6ztM8AAAAAc","level":4,"time":"2018-11-16T12:27:25+00:00","remoteAddr":"213.127.150.1","user":"xxxxx","app":"webdav","method":"REPORT","url":"\/remote.php\/dav\/addressbooks\/users\/xxxxx\/Contactpersonen\/","message":"Exception: HTTP\/1.1 403 Invalid or unknown sync token: {\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\InvalidSyncToken\",\"Message\":\"Invalid or unknown sync token\",\"Code\":0,\"Trace\":\"#0 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Sync\\\/Plugin.php(65): Sabre\\\\DAV\\\\Sync\\\\Plugin->syncCollection('addressbooks\\\/us...', Object(Sabre\\\\DAV\\\\Xml\\\\Request\\\\SyncCollectionReport))\\n#1 [internal function]: Sabre\\\\DAV\\\\Sync\\\\Plugin->Sabre\\\\DAV\\\\Sync\\\\{closure}('{DAV:}sync-coll...', Object(Sabre\\\\DAV\\\\Xml\\\\Request\\\\SyncCollectionReport), 'addressbooks\\\/us...')\\n#2 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Object(Closure), Array)\\n#3 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(718): Sabre\\\\Event\\\\EventEmitter->emit('report', Array)\\n#4 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpReport(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#5 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#6 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(479): Sabre\\\\Event\\\\EventEmitter->emit('method:REPORT', Array)\\n#7 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(254): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#8 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/apps\\\/dav\\\/lib\\\/Server.php(298): Sabre\\\\DAV\\\\Server->exec()\\n#9 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/apps\\\/dav\\\/appinfo\\\/v2\\\/remote.php(31): OCA\\\\DAV\\\\Server->exec()\\n#10 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/remote.php(165): require_once('\\\/home\\\/xxxxx\\\/dom...')\\n#11 {main}\",\"File\":\"\\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Sync\\\/Plugin.php\",\"Line\":123}"}
{"reqId":"W@63ri0s2qdTNtrs6ztM8wAAAAA","level":4,"time":"2018-11-16T12:27:27+00:00","remoteAddr":"213.127.150.1","user":"xxxxx","app":"webdav","method":"REPORT","url":"\/remote.php\/dav\/addressbooks\/users\/xxxxx\/Contactpersonen\/","message":"Exception: HTTP\/1.1 403 Invalid or unknown sync token: {\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\InvalidSyncToken\",\"Message\":\"Invalid or unknown sync token\",\"Code\":0,\"Trace\":\"#0 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Sync\\\/Plugin.php(65): Sabre\\\\DAV\\\\Sync\\\\Plugin->syncCollection('addressbooks\\\/us...', Object(Sabre\\\\DAV\\\\Xml\\\\Request\\\\SyncCollectionReport))\\n#1 [internal function]: Sabre\\\\DAV\\\\Sync\\\\Plugin->Sabre\\\\DAV\\\\Sync\\\\{closure}('{DAV:}sync-coll...', Object(Sabre\\\\DAV\\\\Xml\\\\Request\\\\SyncCollectionReport), 'addressbooks\\\/us...')\\n#2 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Object(Closure), Array)\\n#3 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(718): Sabre\\\\Event\\\\EventEmitter->emit('report', Array)\\n#4 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpReport(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#5 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#6 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(479): Sabre\\\\Event\\\\EventEmitter->emit('method:REPORT', Array)\\n#7 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(254): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#8 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/apps\\\/dav\\\/lib\\\/Server.php(298): Sabre\\\\DAV\\\\Server->exec()\\n#9 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/apps\\\/dav\\\/appinfo\\\/v2\\\/remote.php(31): OCA\\\\DAV\\\\Server->exec()\\n#10 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/remote.php(165): require_once('\\\/home\\\/xxxxx\\\/dom...')\\n#11 {main}\",\"File\":\"\\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Sync\\\/Plugin.php\",\"Line\":123}"}
{"reqId":"W@63wy0s2qdTNtrs6ztNHwAAABI","level":4,"time":"2018-11-16T12:27:47+00:00","remoteAddr":"213.127.150.1","user":"xxxxx","app":"webdav","method":"REPORT","url":"\/remote.php\/dav\/addressbooks\/users\/xxxxx\/Contactpersonen\/","message":"Exception: HTTP\/1.1 403 Invalid or unknown sync token: {\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\InvalidSyncToken\",\"Message\":\"Invalid or unknown sync token\",\"Code\":0,\"Trace\":\"#0 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Sync\\\/Plugin.php(65): Sabre\\\\DAV\\\\Sync\\\\Plugin->syncCollection('addressbooks\\\/us...', Object(Sabre\\\\DAV\\\\Xml\\\\Request\\\\SyncCollectionReport))\\n#1 [internal function]: Sabre\\\\DAV\\\\Sync\\\\Plugin->Sabre\\\\DAV\\\\Sync\\\\{closure}('{DAV:}sync-coll...', Object(Sabre\\\\DAV\\\\Xml\\\\Request\\\\SyncCollectionReport), 'addressbooks\\\/us...')\\n#2 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Object(Closure), Array)\\n#3 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(718): Sabre\\\\Event\\\\EventEmitter->emit('report', Array)\\n#4 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpReport(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#5 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#6 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(479): Sabre\\\\Event\\\\EventEmitter->emit('method:REPORT', Array)\\n#7 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(254): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#8 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/apps\\\/dav\\\/lib\\\/Server.php(298): Sabre\\\\DAV\\\\Server->exec()\\n#9 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/apps\\\/dav\\\/appinfo\\\/v2\\\/remote.php(31): OCA\\\\DAV\\\\Server->exec()\\n#10 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/remote.php(165): require_once('\\\/home\\\/xxxxx\\\/dom...')\\n#11 {main}\",\"File\":\"\\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Sync\\\/Plugin.php\",\"Line\":123}"}
```

3. After opening Contacts app:
```
{"reqId":"W@64dS0s2qdTNtrs6ztOUQAAAD8","level":0,"time":"2018-11-16T12:30:45+00:00","remoteAddr":"213.127.150.1","user":"--","app":"webdav","method":"OPTIONS","url":"\/remote.php\/dav\/principals\/users\/xxxxx","message":"Exception: HTTP\/1.1 401 No public access to this resource., No 'Authorization: Basic' header found. Either the client didn't send one, or the server is misconfigured: {\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\NotAuthenticated\",\"Message\":\"No public access to this resource., No 'Authorization: Basic' header found. Either the client didn't send one, or the server is misconfigured\",\"Code\":0,\"Trace\":\"#0 [internal function]: Sabre\\\\DAV\\\\Auth\\\\Plugin->beforeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#1 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#2 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(466): Sabre\\\\Event\\\\EventEmitter->emit('beforeMethod', Array)\\n#3 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(254): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#4 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/apps\\\/dav\\\/lib\\\/Server.php(298): Sabre\\\\DAV\\\\Server->exec()\\n#5 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/apps\\\/dav\\\/appinfo\\\/v2\\\/remote.php(31): OCA\\\\DAV\\\\Server->exec()\\n#6 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/remote.php(165): require_once('\\\/home\\\/xxxxx\\\/dom...')\\n#7 {main}\",\"File\":\"\\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Auth\\\/Plugin.php\",\"Line\":168}"}
{"reqId":"W@64di0s2qdTNtrs6ztOVQAAACE","level":0,"time":"2018-11-16T12:30:47+00:00","remoteAddr":"213.127.150.1","user":"--","app":"webdav","method":"PROPFIND","url":"\/remote.php\/dav\/addressbooks\/users\/xxxxx\/","message":"Exception: HTTP\/1.1 401 No public access to this resource., No 'Authorization: Basic' header found. Either the client didn't send one, or the server is misconfigured: {\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\NotAuthenticated\",\"Message\":\"No public access to this resource., No 'Authorization: Basic' header found. Either the client didn't send one, or the server is misconfigured\",\"Code\":0,\"Trace\":\"#0 [internal function]: Sabre\\\\DAV\\\\Auth\\\\Plugin->beforeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#1 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#2 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(466): Sabre\\\\Event\\\\EventEmitter->emit('beforeMethod', Array)\\n#3 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(254): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#4 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/apps\\\/dav\\\/lib\\\/Server.php(298): Sabre\\\\DAV\\\\Server->exec()\\n#5 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/apps\\\/dav\\\/appinfo\\\/v2\\\/remote.php(31): OCA\\\\DAV\\\\Server->exec()\\n#6 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/remote.php(165): require_once('\\\/home\\\/xxxxx\\\/dom...')\\n#7 {main}\",\"File\":\"\\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Auth\\\/Plugin.php\",\"Line\":168}"}
{"reqId":"W@64dy0s2qdTNtrs6ztOVwAAADo","level":4,"time":"2018-11-16T12:30:48+00:00","remoteAddr":"213.127.150.1","user":"xxxxx","app":"webdav","method":"REPORT","url":"\/remote.php\/dav\/addressbooks\/users\/xxxxx\/Contactpersonen\/","message":"Exception: HTTP\/1.1 403 Invalid or unknown sync token: {\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\InvalidSyncToken\",\"Message\":\"Invalid or unknown sync token\",\"Code\":0,\"Trace\":\"#0 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Sync\\\/Plugin.php(65): Sabre\\\\DAV\\\\Sync\\\\Plugin->syncCollection('addressbooks\\\/us...', Object(Sabre\\\\DAV\\\\Xml\\\\Request\\\\SyncCollectionReport))\\n#1 [internal function]: Sabre\\\\DAV\\\\Sync\\\\Plugin->Sabre\\\\DAV\\\\Sync\\\\{closure}('{DAV:}sync-coll...', Object(Sabre\\\\DAV\\\\Xml\\\\Request\\\\SyncCollectionReport), 'addressbooks\\\/us...')\\n#2 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Object(Closure), Array)\\n#3 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(718): Sabre\\\\Event\\\\EventEmitter->emit('report', Array)\\n#4 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpReport(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#5 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#6 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(479): Sabre\\\\Event\\\\EventEmitter->emit('method:REPORT', Array)\\n#7 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(254): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#8 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/apps\\\/dav\\\/lib\\\/Server.php(298): Sabre\\\\DAV\\\\Server->exec()\\n#9 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/apps\\\/dav\\\/appinfo\\\/v2\\\/remote.php(31): OCA\\\\DAV\\\\Server->exec()\\n#10 \\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/remote.php(165): require_once('\\\/home\\\/xxxxx\\\/dom...')\\n#11 {main}\",\"File\":\"\\\/home\\\/xxxxx\\\/domains\\\/xxx.xxxxx.xx\\\/public_html\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Sync\\\/Plugin.php\",\"Line\":123}"}
```

After first successful sync, the sync does not work anymore! Even after changing a contact in ownCloud contacts, those changes are not synced to macOS Contacts.

I hope this helps...

**Comment by joostdekeijzer** [comment]
Ok, final results of my research of today:

On macOS Mojave (10.14) the sync system issues the following for the REPORT http request:
```
<?xml version="1.0" encoding="UTF-8"?>
<A:sync-collection xmlns:A="DAV:">
  <A:sync-token>b905a5fd-ce23-4190-9be3-236969d071a8</A:sync-token>
  <A:sync-level>1</A:sync-level>
  <A:prop>
    <A:getetag/>
  </A:prop>
</A:sync-collection>
```

The cardDAV sync agent on my Android phone issues the following (which is supported)
```
<?xml version="1.0" encoding="utf-8" ?>
<D:sync-collection xmlns:D="DAV:">
<D:sync-token>http://sabre.io/ns/sync/2285</D:sync-token>
<D:sync-level>1</D:sync-level>
<D:prop>
<D:getcontenttype />
<D:getetag />
</D:prop>
</D:sync-collection>
```

So it seems in Mojave Apple has changed the way it requests REPORT statements and thus breaks the sync?

**Comment by Julian1998** [comment]
@joostdekeijzer It's the same with CalDav

**Comment by Julian1998** [comment]
I found out several weeks ago that Apple doesn't send credentials when using http

**Comment by Julian1998** [comment]
Are you guys using http or https?

**Comment by joostdekeijzer** [comment]
Just filed a bug report with Apple (#46163430) referencing this post.

**Comment by DeepDiver1975** [comment]
> Just filed a bug report with Apple (#46163430) referencing this post.

Awesome - thanks a lot! Please keep us in the loop so that we can find a solution together. THX

**Comment by Julian1998** [comment]
@joostdekeijzer Is it possible for you to duplicate your issue on openradar so we can see what's happening? Thanks in advance!

**Comment by joostdekeijzer** [comment]
@Julian1998 you mean http://openradar.appspot.com/ ?

**Comment by Julian1998** [comment]
@joostdekeijzer Exactly!

**Comment by joostdekeijzer** [comment]
Here you go: http://openradar.appspot.com/radar?id=4942230217293824

**Comment by Julian1998** [comment]
@joostdekeijzer Thanks for sharing infos!

**Comment by crimsen** [comment]
Is this bug already fixed? In openradar is this bug still open.

**Comment by joostdekeijzer** [comment]
My ticket with Apple is also still open. No real hopes  this will  change btw.

**Comment by DeepDiver1975** [comment]
🤖 Automated issue analysis by Claude Code review agent.

**Classification: confirmed-bug** (client-interop bug, server-side workaround feasible)

## Summary

This 2018 report ("CardDAV always returns a 403 saying sync token is invalid or unknown") is a genuine, reproducible interoperability bug, and it remains **open and unresolved**. The root cause was correctly diagnosed within the thread by an ownCloud maintainer and is well understood; what is missing is a server-side workaround.

## What is actually happening

The `403 Invalid or unknown sync token` originates in the vendored sabre/dav library, not in ownCloud's own code:

```
Sabre\DAV\Exception\InvalidSyncToken
  at lib/composer/sabre/dav/lib/DAV/Sync/Plugin.php:123
```

Sabre issues sync-tokens in the opaque URI form it controls, e.g.:

```
<D:sync-token>http://sabre.io/ns/sync/2285</D:sync-token>
```

`Sync/Plugin.php::syncCollection()` validates an incoming `sync-token` by stripping its own `SYNCTOKEN_PREFIX` (`http://sabre.io/ns/sync/`). Any token not bearing that prefix is rejected as invalid.

The bug is triggered because macOS (10.13 High Sierra / 10.14 Mojave) Contacts sends a **bare UUID** instead of the server-issued URI token:

```xml
<A:sync-collection xmlns:A="DAV:">
  <A:sync-token>b905a5fd-ce23-4190-9be3-236969d071a8</A:sync-token>
  <A:sync-level>1</A:sync-level>
  <A:prop><A:getetag/></A:prop>
</A:sync-collection>
```

(reporter joostdekeijzer, https://github.com/owncloud/core/issues/30820#issuecomment-439454964). The same client over Android/DAVx5 correctly echoes back `http://sabre.io/ns/sync/NNNN` and syncs fine.

This violates **RFC 6578 §4**, which requires the client to return the sync-token exactly as the server provided it. ownCloud maintainer @DeepDiver1975 confirmed this (https://github.com/owncloud/core/issues/30820#issuecomment-439807951):

> This is the issue — no idea where Apple is taking the sync token from. Sync tokens are generated by us (sabre dav) in the format `http://sabre.io/ns/sync/2285`. Any other synctoken cannot be used by us. As per RFC 6578 the sync token has to be a valid URI and has to be used as returned in the sync collection.

A bug was filed with Apple (#46163430, marked duplicate of 42566961). As of the last comment (2020-05-11) the Apple radar and this issue are both still open, and the reporter held out "no real hopes this will change."

## Why this is classified confirmed-bug rather than invalid/upstream-only

- **Reproducible and multi-user confirmed**: independent reports on ownCloud 10.0.9 / 10.0.10 across macOS 10.13.4–10.14.1 (robertoschwald, Perm1990, joostdekeijzer).
- **Not obsolete**: `owncloud/core` is still an actively maintained codebase (commits dated 2026-06-19 on `master`), so this is not stale due to project end-of-life.
- **No fix has landed**: nothing in the thread or subsequent activity indicates a workaround was merged. The error path is unchanged.
- While the *trigger* is an Apple client deviating from RFC 6578, the symptom is a hard `403` that breaks contact sync for affected users, and a safe server-side mitigation is demonstrably available (a community member already prototyped one in-thread). This makes it actionable on the ownCloud side rather than purely "won't fix / report to vendor."

## Proposed fix

**Root cause:** sabre/dav rejects any `sync-token` that does not carry its `http://sabre.io/ns/sync/` prefix; the macOS client sends a bare UUID, producing `InvalidSyncToken` (403).

**Strategy:** When an incoming sync-token is clearly not a sabre-issued token (e.g. a bare UUID, or more generally any value lacking the known prefix), treat it as a *first/initial* sync request — i.e. coerce it to `null` so the server returns the full collection plus a fresh, valid sync-token. The client then stores and re-uses the correct token. This is exactly the behavior sabre already implements for a missing token, so it degrades gracefully (one full re-sync) instead of failing hard. This mirrors the in-thread workaround by joostdekeijzer (https://github.com/owncloud/core/issues/30820#issuecomment-439625028).

**Where to implement (preferred → least preferred):**

1. **`apps/dav/` server middleware / plugin (recommended).** Do not patch the vendored `lib/composer/sabre/dav/...` file (composer-managed; lost on `composer update`). Instead register a small plugin or `beforeMethod`/REPORT hook in ownCloud's own DAV stack (around `OCA\DAV\Server` / `apps/dav/lib/`) that inspects the incoming `{DAV:}sync-collection` REPORT body and, if the `sync-token` does not start with `http://sabre.io/ns/sync/` (and matches a UUID-shaped value), rewrites/clears it before sabre's `Sync\Plugin` runs. Gate it behind a config flag (e.g. `dav.tolerate_foreign_sync_tokens`, default true) so the relaxed validation is auditable.

2. **Upstream sabre/dav.** The cleanest long-term home is sabre/dav itself (make `Sync\Plugin::syncCollection()` fall back to a full sync on an unrecognized token rather than throw). Worth a linked upstream issue/PR, but ownCloud can ship the app-level mitigation immediately without waiting on upstream.

**Files likely involved:**
- `apps/dav/lib/Server.php` (where the sabre `Server` and plugins are assembled) — register the mitigation plugin here.
- New file, e.g. `apps/dav/lib/DAV/SyncTokenCompatibilityPlugin.php` — the hook implementation.
- Reference (read-only, do **not** edit): `lib/composer/sabre/dav/lib/DAV/Sync/Plugin.php:119-123` (`SYNCTOKEN_PREFIX`, throw site).

**Test approach:**
- Unit test the new plugin: feed a `SyncCollectionReport` with a bare-UUID token and assert the token is normalized to `null` (full sync), and that a valid `http://sabre.io/ns/sync/N` token is passed through untouched.
- Integration/functional test against the DAV endpoint: POST a Mojave-style `REPORT` body (bare UUID sync-token) to `/remote.php/dav/addressbooks/users/<user>/<book>/` and assert a `207 Multi-Status` full listing with a fresh valid sync-token in the response, instead of `403 InvalidSyncToken`.
- Regression: confirm normal incremental sync (valid prefixed token) still returns only deltas, and that an explicitly *malformed-but-prefixed* token still yields the standard 403.
- Manual: re-add a CardDAV account in macOS Contacts and confirm contacts sync bidirectionally after the initial full sync.

## Recommended issue actions

- Post this analysis as a comment.
- Mention @DeepDiver1975 (original maintainer who diagnosed the root cause) to confirm whether an app-level mitigation is acceptable or whether ownCloud's position remains "client/Apple bug, won't work around."
- Keep open: it is a confirmed, still-reproducible bug with an actionable server-side mitigation.

---
*Read-only automated analysis. No repository writes were performed by this agent.*

**Comment by DeepDiver1975** [comment]
closing this - I have no hope ....
